### ISSUE TITLE
Dashboard filters should not reference join aliases

### ISSUE THREAD

**Issue by ZeRego** [lightdash/lightdash#12680]
### Description

tldr: Dashboard filters should not use fields from joined tables. The field options should all be from the main tables/explores.

---
## Problem:

The autocomplete endpoint relies on the **tableName** and **fieldId**. From the **tableName** we try to find the explore with the same name and as a fallback we try to find the first explore which has that **tableName** as a joined table.
The fallback is a lottery system if you have multiple explores using the same alias for different tables.

Example:
They have a dashboard filter for a products field.

Get endpoint params:
```
tableName = 'products'
fieldId = 'products_product_name'
```

There is no explore named "products" so it finds a fallback.
The first explore that joins "products" is "product_rules_customer_product_requests".

Join:
```
{
        "sqlOn": "${product_rules_customer_product_requests.product_id} = ${products.product_id}",
        "table": "products",
        "compiledSqlOn": "(\"product_rules_customer_product_requests\".product_id) = (\"products\".product_id)"
    }
```

Table:
```
"products": {
            "name": "products",
            "label": "Products",
            "schema": "DBT_PROD_mart",
            "database": "DW_PROD",
            "sqlTable": "DW_PROD.DBT_PROD_mart.core_products",
            "groupLabel": "Products & Brands",
            "description": "This table contains every products with unique identifiers and a creation/updated date.",
            "groupDetails": {},
            "originalName": "core_products",
            "orderFieldsBy": "LABEL",
            "requiredFilters": []
        }
```

The SQL query will use a "product_rules_customer_product_requests"  and join "products" instead of using the explore "core_products". 
This can cause unwanted side effects. Eg:
- mandatory filters / row level filtering
- inner joins

## Solution

Dashboard filters should not use fields from joined tables. The options should all be from the main tables/explores.

Example:

The available fields for filters would be from table "core_products" rather than "products".

Get endpoint params:
tableName = 'core_products'
fieldId = 'core_products_product_name'

Find explore
We find an explore with that name otherwise we throw an error. There is no fallback.

To join other filters, we would have to map the original table names with joined alias.
eg: "cached_explores_id" match joined "explores_id"

### Steps to Reproduce the Bug or Issue

1. create explores A, B, C and D
2. explore C joins A with alias "test"
3. explore D joins B with alias "test"
4. create charts and dashboard
5. add filter for "test" dimension

Current:
dashboard shows filter options for "test". 
Autocomplete runs a SQL for explore C/D (depending on which one is first in the DB)

Expected:
dashboard shows filter options for A and B. 
Autocomplete runs a SQL for explore A/B.

### Cloud or self-hosting

None

**Comment by yadavshubham01** [comment]
hey @ZeRego Is this issue still open ??

**Comment by ZeRego** [comment]
This ticket is not ideal for external contributors as it contains a lot of context that is hard to share and will likely involve breaking changes or data migration.
